src/prefect/orion/schemas/core.py

```diff
@@ -436,6 +436,14 @@ class TaskRun(ORMBaseModel):
         default=None, description="The current task run state."
     )
 
+    @validator("cache_key", pre=True)
+    def validate_cache_key_length(cls, cache_key):
+        if len(cache_key) > PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH.value():
+            raise ValueError(
+                f"Invalid cache_key length. Task cache key length can be modified using the PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH setting."
```

Comment:
Can we clarify that this is a server-side setting?

Reply:
Shortened to "Cache key exceeded maximum allowed length of {PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH} characters."